### ISSUE TITLE
Bug: Internal Server Error for Go hello-world template

### ISSUE THREAD

**Issue by mikepianka** [aws/aws-sam-cli-app-templates#421]
### Description:
The Go hello-world template returns an "Internal Server Error" response.

### Steps to reproduce:
- On Windows 11 install Docker Desktop and the SAM CLI
- Use `sam init` with the Go 1.x hello world template
- Use `make` to install deps
- Use `sam local start-api` to run the local emulator
- Hit `http://localhost:3000/hello`

### Observed result:
There is an "Internal Server Error" response and the following TLS error is thrown in the console:

```shell
2023-05-22 09:20:37,983 | Found one Lambda function with name 'HelloWorldFunction'                                                                                                                     
2023-05-22 09:20:37,983 | Invoking hello-world (go1.x)                                                                                                                                                 
2023-05-22 09:20:37,984 | Loading AWS credentials from session with profile 'None'                                                                                                                     
2023-05-22 09:20:39,998 | Resolving code path. Cwd=/home/mike/github_locals/sam-presigned-urls/.aws-sam/build, CodeUri=/home/mike/github_locals/sam-presigned-urls/.aws-sam/build/HelloWorldFunction   
2023-05-22 09:20:39,999 | Resolved absolute path to code is /home/mike/github_locals/sam-presigned-urls/.aws-sam/build/HelloWorldFunction                                                              
2023-05-22 09:20:40,005 | Reuse the created warm container for Lambda function 'HelloWorldFunction'                                                                                                    
2023-05-22 09:20:40,010 | Lambda function 'HelloWorldFunction' is already running                                                                                                                      
2023-05-22 09:20:40,012 | Starting a timer for 5 seconds for function 'HelloWorldFunction'                                                                                                             
START RequestId: 783be961-9d09-46fb-be79-c707dd67607c Version: $LATEST
Get "https://checkip.amazonaws.com": tls: failed to verify certificate: x509: certificate is valid for *.safezone.mcafee.com, safezone.mcafee.com, not checkip.amazonaws.com: Error
null
2023-05-22 09:20:40,183 | Unable to find Click Context for getting session_id.                                                                                                                         
END RequestId: 783be961-9d09-46fb-be79-c707dd67607c
REPORT RequestId: 783be961-9d09-46fb-be79-c707dd67607c  Init Duration: 0.10 ms  Duration: 167.05 ms     Billed Duration: 168 ms Memory Size: 128 MB     Max Memory Used: 128 MB
2023-05-22 09:20:40,184 | Lambda returned empty body!                                                                                                                                                  
2023-05-22 09:20:40,185 | Invalid lambda response received: Invalid API Gateway Response Keys: {'errorType', 'errorMessage'} in {'errorMessage': 'Get "https://checkip.amazonaws.com": tls: failed to  
verify certificate: x509: certificate is valid for *.safezone.mcafee.com, safezone.mcafee.com, not checkip.amazonaws.com', 'errorType': 'Error'}                                                       
2023-05-22 09:20:40 127.0.0.1 - - [22/May/2023 09:20:40] "GET /hello HTTP/1.1" 502 -
```

### Expected result:
Hello world response and no error.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Windows 11
2. If using SAM CLI, `sam --version`: SAM CLI, version 1.84.0
3. AWS region: us-east-1

`Add --debug flag to any SAM CLI commands you are running`

**Comment by moelasmar** [comment]
Thanks @mikepianka for raising this issue.
Unfortunately I am not able to reproduce this issue, and I think the problem is I do not have a windows machine with the same setup you have. It seems you have mcafee firewall installed in your machine, and it is blocking the call to `https://checkip.amazonaws.com`

**Comment by moelasmar** [comment]
can you try to allow list the url `https://checkip.amazonaws.com` in your Mcafee

**Comment by mikepianka** [comment]
So I'm not actually running Mcafee and I did think that error was strange. I think the issue might actually just be with the https://checkip.amazonaws.com endpoint itself. It seems like its TLS cert is flakey. 

I just tried again with curl, and I was able to successfully hit the endpoint, but then a few minutes later I tried again and hit an error. In Chrome, I tried visiting it and got redirected to my ISP's page with a Mcafee warning (that must be where the Mcafee message originates from). I tried refreshing in Chrome a few minutes later and it went through and showed my IP as expected.

![image](https://github.com/aws/aws-sam-cli-app-templates/assets/47948499/9495d7a7-0323-4288-a5bd-2d5eab9293b8)

I changed up the hello world example to use jsonplaceholder and it all works as expected.

```go
package main

import (
	"encoding/json"
	"errors"
	"fmt"
	"io"
	"net/http"

	"github.com/aws/aws-lambda-go/events"
	"github.com/aws/aws-lambda-go/lambda"
)

var (
	// DefaultHTTPGetAddress Default Address
	DefaultHTTPGetAddress = "https://jsonplaceholder.typicode.com/albums/1"

	// ErrNoTitle No album title found in response
	ErrNoTitle = errors.New("No album title in HTTP response")

	// ErrNon200Response non 200 status code in response
	ErrNon200Response = errors.New("Non 200 Response found")
)

type Album struct {
	UserID int    `json:"userId"`
	ID     int    `json:"id"`
	Title  string `json:"title"`
}

func handler(request events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
	resp, err := http.Get(DefaultHTTPGetAddress)
	if err != nil {
		return events.APIGatewayProxyResponse{}, err
	}

	if resp.StatusCode != 200 {
		return events.APIGatewayProxyResponse{}, ErrNon200Response
	}

	content, err := io.ReadAll(resp.Body)
	if err != nil {
		return events.APIGatewayProxyResponse{}, err
	}

	var album Album

	err = json.Unmarshal(content, &album)
	if err != nil {
		return events.APIGatewayProxyResponse{}, err
	}

	if len(album.Title) == 0 {
		return events.APIGatewayProxyResponse{}, ErrNoTitle
	}

	return events.APIGatewayProxyResponse{
		Body:       fmt.Sprintf("Hello! Have you listened to the new album <%s>?\n", album.Title),
		StatusCode: 200,
	}, nil
}

func main() {
	lambda.Start(handler)
}

```

![image](https://github.com/aws/aws-sam-cli-app-templates/assets/47948499/a4b56758-8c2b-4a14-9673-56ce538bbf2d)

**Comment by moelasmar** [comment]
thanks @mikepianka for the detailed explanation.  As you mentioned, the problem is in `https://checkip.amazonaws.com/` endpoint itself, which we could not fix. But I believe it is better to update our template to use some other endpoint or something else, as using this endpoint is not the purpose of our templates.

**Comment by mikepianka** [comment]
Sounds good. Do you want me to open a PR with my example? I could swap it out for a different endpoint if you have something else in mind.

**Comment by mikepianka** [comment]
Actually I just realized that since this example uses API Gateway proxy, you should be able to just get the caller's IP right from the identity sourceIP property on the request object. So the example could function the same but not rely on an extra web request. What do you think of that?

**Comment by moelasmar** [comment]
Actually the main purpose of this template is to show how to create a Go Lambda function with an API Gateway resource. Calling `https://checkip.amazonaws.com` endpoint or any other service is not part of the template purpose. So, I think it will be better to use the Request to get this IP,  but make sure that if the Lambda function got invoked directly with different event, it will not fail. Your contribution is appreciated :)

**Comment by mikepianka** [comment]
Hi @moelasmar. It took a bit of time for me to get back to this, but I made a PR to fix the examples here - https://github.com/aws/aws-sam-cli-app-templates/pull/431

**Comment by mildaniel** [comment]
This has been merged in #431, resolving.
